### ISSUE TITLE
args library initialization

### ISSUE THREAD

**Issue by tegoo** [legatoproject/legato-af#4]
Hey, guys!

Found one more potential problem. You use legato shared library initialization routine to init all runtime libraries:
```
__attribute__((constructor)) void _legato_InitFramework
(
    int argc,       ///< Number of command-line arguments (including command name)
    char* argv[]    ///< List of command-line arguments (argv[0] is the name of the command).
)
{
    // Gather the program arguments for later processing.
    arg_SetArgs((size_t)argc, (char**)argv);

    // The order of initialization is important.  Ideally, logging would be initialized first,
    // because before that, any logging calls will report the wrong component.  However, the
    // logging uses memory pools, so memory pools must be initialized before logging.

    mem_Init();
    log_Init();        // Uses memory pools.
    sig_Init();        // Uses memory pools.
    safeRef_Init();    // Uses memory pools and hash maps.
    pathIter_Init();   // Uses memory pools and safe references.
    mutex_Init();      // Uses memory pools.
    sem_Init();        // Uses memory pools.
    thread_Init();     // Uses memory pools and safe references.
    event_Init();      // Uses thread API.
    timer_Init();      // Uses event loop.
    msg_Init();        // Uses event loop.
    kill_Init();       // Uses memory pools and timers.
    properties_Init(); // Uses memory pools and safe references.

    // This must be called last, because it calls several subsystems to perform the
    // thread-specific initialization for the main thread.
    thread_InitThread();
}
```

It pretty executed during dynamic library loading. But to pass or not `int argc,  char* argv[]` arguments is up to compiler. For instance, my compiler included to cross toolchain built with YOCTO does not passes it. So, any invocation of any cmd tool results in error because no arguments even if I pass arguments. 

I think the most suitable fix is to add `arg_SetArgs((size_t)argc, (char**)argv);` invocation to code generation routine:

```
void GenerateExeMain
(
    const model::Exe_t* exePtr,
    const mk::BuildParams_t& buildParams
)
{
    // ...
    // Define main().
                  "int main(int argc, char* argv[])\n"
                  "{\n"
                  "    arg_SetArgs((size_t)argc, (char**)argv);"

    // Make stdout line buffered.
                  "    // Make stdout line buffered so printf shows up in logs without flushing.\n"
                  "    setlinebuf(stdout);\n"
                  "\n"
    //...
}
```

**Comment by jtchitty** [comment]
I'm going to fix this now.  Thanks for pointing it out, tegoo.  We'd spotted the problem before, but it seemed to be working for all the cases we saw on a regular basis, so the priority has been lower than it probably should have been.

I agree with your proposed fix too, by the way.

Cheers!

**Comment by CoRfr** [comment]
Issue has been addressed in 16.04.0
